Source/Core/DolphinQt2/Config/ControllersWindow.cpp

```diff
@@ -0,0 +1,351 @@
+      wm_box->addItem(item);
+
+    wm_group->addWidget(wm_box);
+    wm_group->addWidget(wm_button);
+
+    m_wiimote_layout->addRow(tr("Wii Remote %1").arg(i + 1), wm_group);
+  }
+
+  m_wiimote_layout->addRow(m_wiimote_continous_scanning);
+  m_wiimote_layout->addRow(m_wiimote_real_balance_board);
+  m_wiimote_layout->addRow(m_wiimote_speaker_data);
+
+  m_wiimote_box->setLayout(m_wiimote_layout);
+}
+
+void ControllersWindow::CreateAdvancedLayout()
+{
+  m_advanced_box = new QGroupBox(tr("Advanced"));
+  m_advanced_layout = new QHBoxLayout();
+  m_advanced_bg_input = new QCheckBox(tr("Background Input"));
+
+  m_advanced_layout->addWidget(m_advanced_bg_input);
+
+  m_advanced_box->setLayout(m_advanced_layout);
+}
+
+void ControllersWindow::CreateMainLayout()
+{
+  m_main_layout = new QVBoxLayout();
+  m_main_hbox = new QHBoxLayout();
+
+  m_main_hbox->addWidget(m_gc_box);
+  m_main_hbox->addWidget(m_wiimote_box);
+
+  m_main_layout->addItem(m_main_hbox);
+  m_main_layout->addWidget(m_advanced_box);
+
+  setLayout(m_main_layout);
+}
+
+void ControllersWindow::ConnectWidgets()
+{
+  connect(m_wiimote_passthrough, SIGNAL(toggled(bool)), this, SLOT(OnWiimoteModeChanged(bool)));
+
+  connect(m_advanced_bg_input, SIGNAL(toggled(bool)), this, SLOT(SaveSettings()));
+  connect(m_wiimote_continous_scanning, SIGNAL(toggled(bool)), SLOT(SaveSettings()));
+  connect(m_wiimote_real_balance_board, SIGNAL(toggled(bool)), SLOT(SaveSettings()));
+  connect(m_wiimote_speaker_data, SIGNAL(toggled(bool)), SLOT(SaveSettings()));
+  connect(m_wiimote_sync, SIGNAL(clicked()), SLOT(OnBluetoothPassthroughSyncPressed()));
+  connect(m_wiimote_reset, SIGNAL(clicked()), SLOT(OnBluetoothPassthroughResetPressed()));
+
+  for (int i = 0; i < 4; i++)
+  {
+    connect(m_wiimote_boxes[i], SIGNAL(currentIndexChanged(int)), SLOT(SaveSettings()));
+    connect(m_wiimote_boxes[i], SIGNAL(currentIndexChanged(int)), SLOT(OnWiimoteTypeChanged(int)));
+    connect(m_wiimote_buttons[i], SIGNAL(clicked()), SLOT(UnimplementedButton()));
+    connect(m_gc_controller_boxes[i], SIGNAL(currentIndexChanged(int)), SLOT(SaveSettings()));
+    connect(m_gc_controller_boxes[i], SIGNAL(currentIndexChanged(int)), SLOT(OnGCTypeChanged(int)));
+    connect(m_gc_buttons[i], SIGNAL(clicked()), SLOT(UnimplementedButton()));
+  }
+}
+
+void ControllersWindow::OnWiimoteModeChanged(bool passthrough)
+{
+  m_wiimote_sync->setEnabled(passthrough);
+  m_wiimote_reset->setEnabled(passthrough);
+
+  for (int i = 0; i < 4; i++)
+  {
+    m_wiimote_boxes[i]->setEnabled(!passthrough);
+    m_wiimote_buttons[i]->setEnabled(!passthrough);
+  }
+
+  m_wiimote_real_balance_board->setEnabled(!passthrough);
+  m_wiimote_speaker_data->setEnabled(!passthrough);
+  m_wiimote_continous_scanning->setEnabled(!passthrough);
+
+  SaveSettings();
+}
+
+void ControllersWindow::OnWiimoteTypeChanged(int type)
+{
+  auto* box = static_cast<QComboBox*>(QObject::sender());
+
+  for (int i = 0; i < 4; i++)
```

Comment:
I agree with the first two, but every version I try to write only gets more complicated when using ``std::find_if``